ext/opentelemetry-ext-datadog/src/opentelemetry/ext/datadog/__init__.py

```diff
@@ -0,0 +1,211 @@
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+
+    trace.set_tracer_provider(TracerProvider())
+    tracer = trace.get_tracer(__name__)
+
+    exporter = DatadogSpanExporter(
+        agent_url="http://agent:8126", service="my-helloworld-service"
+    )
+
+    span_processor = BatchExportSpanProcessor(exporter)
+    trace.get_tracer_provider().add_span_processor(span_processor)
+
+    with tracer.start_as_current_span("foo"):
+        print("Hello world!")
+
+API
+---
+.. _Datadog: https://www.datadoghq.com/
+.. _OpenTelemetry: https://github.com/open-telemetry/opentelemetry-python/
+"""
+
+import logging
+import os
+from urllib.parse import urlparse
+
+from ddtrace.encoding import MsgpackEncoder
+from ddtrace.internal.writer import AgentWriter
+from ddtrace.span import Span as DatadogSpan
+
+import opentelemetry.trace as trace_api
+from opentelemetry.sdk.trace import Span
+from opentelemetry.sdk.trace.export import SpanExporter, SpanExportResult
+from opentelemetry.trace.status import StatusCanonicalCode
+
+log = logging.getLogger(__name__)
+
+DEFAULT_AGENT_URL = "http://localhost:8126"
+DATADOG_SPAN_TYPES = (
+    "sql",
+    "mongodb",
+    "http",
+)
+
+
+class DatadogSpanExporter(SpanExporter):
+    """Datadog span exporter for OpenTelemetry.
+
+    Args:
+        agent_url: The url of the Datadog Agent or use `DD_TRACE_AGENT_URL` environment variable
+        service: The service to be used for the application or use `DD_SERVICE` environment variable
+    """
+
+    def __init__(self, agent_url=None, service=None):
+        self.agent_url = (
+            agent_url
+            if agent_url
+            else os.environ.get("DD_TRACE_AGENT_URL", DEFAULT_AGENT_URL)
+        )
+        self.service = service if service else os.environ.get("DD_SERVICE")
+        self._agent_writer = None
+
+    @property
+    def agent_writer(self):
+        if self._agent_writer is None:
+            url_parsed = urlparse(self.agent_url)
+            if url_parsed.scheme in ("http", "https"):
+                self._agent_writer = AgentWriter(
+                    hostname=url_parsed.hostname,
+                    port=url_parsed.port,
+                    https=url_parsed.scheme == "https",
+                )
+            elif url_parsed.scheme == "unix":
+                self._agent_writer = AgentWriter(uds_path=url_parsed.path)
+            else:
+                raise ValueError(
+                    "Unknown scheme `%s` for agent URL" % url_parsed.scheme
+                )
+        return self._agent_writer
+
+    def export(self, spans):
+        datadog_spans = self._translate_to_datadog(spans)
+
+        self.agent_writer.write(spans=datadog_spans)
+
+        return SpanExportResult.SUCCESS
+
+    def shutdown(self):
+        if self.agent_writer.started:
+            self.agent_writer.stop()
+            self.agent_writer.join(self.agent_writer.exit_timeout)
```

Comment:
Are the otel folks cool with having a blocking operation like this? It could block for up to a second potentially. Might be worth adding a comment to state that it's required to ensure that the traces are flushed before shutdown. 🙂